src/radiation/geodesics.hpp

```diff
@@ -39,7 +39,34 @@ KOKKOS_INLINE_FUNCTION
 void GetKSource(Real &X0, Real &X1, Real &X2, Real &X3, Real &Kcov0, Real &Kcov1,
                 Real &Kcov2, Real &Kcov3, Real &Kcon0,
                 const Geometry::CoordSysMeshBlock &geom, Real source[4]) {
-  SPACETIMELOOP(mu) { source[mu] = 0.; }
+  Real Kcov[NDFULL] = {Kcov0, Kcov1, Kcov2, Kcov3};
+  Real Xm[NDFULL], Xp[NDFULL];
+  Real Gconm[NDFULL][NDFULL], Gconp[NDFULL][NDFULL];
+  Real dG;
+
+  constexpr Real DELTA = 1.0e-6;
+
+  SPACETIMELOOP(mu) {
+    source[mu] = 0.;
```

Comment:
Maybe just add a `TODO()` here pointing out that we can come back and exploit spacetime symmetries to reduce the compute burden later?

Reply:
done